System controller crashing on site delete (using reload-type: auto)

**Describe the bug**
Deleting a podman site using `skupper --platform podman site delete <name>` with the system controller running using the auto reload type, is crashing the system controller container.

```
2026/06/09 15:36:05 INFO Stopping namespace controller component=namespaces.handler namespace=default
panic: close of closed channel

goroutine 206 [running]:
github.com/skupperproject/skupper/internal/nonkube/controller.(*NamespaceController).Stop(...)
	/go/src/app/internal/nonkube/controller/namespace_controller.go:74
github.com/skupperproject/skupper/internal/nonkube/controller.(*NamespacesHandler).OnRemove(0xc000381920, {0xc000346f40, 0x1a})
	/go/src/app/internal/nonkube/controller/namespaces.go:139 +0x34d
github.com/skupperproject/skupper/internal/filesystem.(*FileWatcher).dispatchTriggers.func1()
	/go/src/app/internal/filesystem/file_watcher.go:151 +0x2b
created by github.com/skupperproject/skupper/internal/filesystem.(*FileWatcher).dispatchTriggers in goroutine 14
	/go/src/app/internal/filesystem/file_watcher.go:150 +0x179
```

**How To Reproduce**
1. Install the system controller using reload type auto: `skupper --platform podman system install --reload-type auto`
3. Create a podman site: `skupper --platform podman site create mysite`
4. Delete the site using: `skupper --platform podman site delete mysite`

**Environment details**
 - Version: v2-dev
 - Platform: podman or docker

--- a/internal/nonkube/controller/event_deduplicator.go
+++ b/internal/nonkube/controller/event_deduplicator.go
@@ -56,6 +56,7 @@ func (ed *EventDeduplicator) processEvents() {
 				timer.Stop()
 			}
 			ed.mutex.Unlock()
+			close(ed.eventCh)
 			return
 		}
 	}
@@ -78,7 +79,6 @@ func (ed *EventDeduplicator) QueueEvent(filename string) {
 }
 
 func (ed *EventDeduplicator) Stop() {
-	if ed.stopCh != nil {
-		close(ed.stopCh)
-	}
+	// eventCh is automatically closed when stopCh is closed
+	// This method is kept for backward compatibility but does nothing
 }

--- a/internal/nonkube/controller/event_deduplicator.go
+++ b/internal/nonkube/controller/event_deduplicator.go
@@ -16,6 +16,7 @@ type EventDeduplicator struct {
 	pendingEvents map[string]*time.Timer
 	handler       func(string)
 	logger        *slog.Logger
+	closeOnce     sync.Once
 }
 
 func NewEventDeduplicator(stopCh chan struct{}, handler func(string), logger *slog.Logger) *EventDeduplicator {
@@ -28,6 +29,7 @@ func NewEventDeduplicator(stopCh chan struct{}, handler func(string), logger *sl
 	}
 
 	go ed.processEvents()
+	go ed.handleStop()
 
 	return ed
 }
@@ -37,7 +39,11 @@ func (ed *EventDeduplicator) processEvents() {
 
 	for {
 		select {
-		case filename := <-ed.eventCh:
+		case filename, ok := <-ed.eventCh:
+			if !ok {
+				return
+			}
+
 			ed.mutex.Lock()
 
 			if timer, exists := ed.pendingEvents[filename]; exists {
@@ -49,15 +55,6 @@ func (ed *EventDeduplicator) processEvents() {
 			})
 
 			ed.mutex.Unlock()
-
-		case <-ed.stopCh:
-			ed.mutex.Lock()
-			for _, timer := range ed.pendingEvents {
-				timer.Stop()
-			}
-			ed.mutex.Unlock()
-			close(ed.eventCh)
-			return
 		}
 	}
 }
@@ -78,7 +75,28 @@ func (ed *EventDeduplicator) QueueEvent(filename string) {
 	}
 }
 
+func (ed *EventDeduplicator) handleStop() {
+	t := time.NewTicker(time.Second)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-ed.stopCh:
+			ed.logger.Info("Parent channel is closed")
+			ed.Stop()
+			return
+		case <-t.C:
+		}
+	}
+}
+
 func (ed *EventDeduplicator) Stop() {
-	// eventCh is automatically closed when stopCh is closed
-	// This method is kept for backward compatibility but does nothing
+	ed.closeOnce.Do(func() {
+		ed.mutex.Lock()
+		for _, timer := range ed.pendingEvents {
+			timer.Stop()
+		}
+		close(ed.eventCh)
+		ed.mutex.Unlock()
+	})
 }

--- a/internal/nonkube/controller/event_deduplicator.go
+++ b/internal/nonkube/controller/event_deduplicator.go
@@ -12,6 +12,7 @@ import (
 type EventDeduplicator struct {
 	eventCh       chan string
 	stopCh        chan struct{}
+	done          chan struct{}
 	mutex         sync.Mutex
 	pendingEvents map[string]*time.Timer
 	handler       func(string)
@@ -23,13 +24,13 @@ func NewEventDeduplicator(stopCh chan struct{}, handler func(string), logger *sl
 	ed := &EventDeduplicator{
 		eventCh:       make(chan string, 10),
 		stopCh:        stopCh,
+		done:          make(chan struct{}),
 		pendingEvents: make(map[string]*time.Timer),
 		handler:       handler,
 		logger:        logger,
 	}
 
 	go ed.processEvents()
-	go ed.handleStop()
 
 	return ed
 }
@@ -39,11 +40,15 @@ func (ed *EventDeduplicator) processEvents() {
 
 	for {
 		select {
-		case filename, ok := <-ed.eventCh:
-			if !ok {
-				return
-			}
+		case <-ed.stopCh:
+			ed.logger.Info("Parent channel is closed")
+			ed.Stop()
+			return
 
+		case <-ed.done:
+			return
+
+		case filename := <-ed.eventCh:
 			ed.mutex.Lock()
 
 			if timer, exists := ed.pendingEvents[filename]; exists {
@@ -75,28 +80,14 @@ func (ed *EventDeduplicator) QueueEvent(filename string) {
 	}
 }
 
-func (ed *EventDeduplicator) handleStop() {
-	t := time.NewTicker(time.Second)
-	defer t.Stop()
-
-	for {
-		select {
-		case <-ed.stopCh:
-			ed.logger.Info("Parent channel is closed")
-			ed.Stop()
-			return
-		case <-t.C:
-		}
-	}
-}
-
 func (ed *EventDeduplicator) Stop() {
 	ed.closeOnce.Do(func() {
 		ed.mutex.Lock()
 		for _, timer := range ed.pendingEvents {
 			timer.Stop()
 		}
-		close(ed.eventCh)
 		ed.mutex.Unlock()
+
+		close(ed.done)
 	})
 }

--- a/internal/nonkube/controller/event_deduplicator_test.go
+++ b/internal/nonkube/controller/event_deduplicator_test.go
@@ -24,7 +24,7 @@ func TestEventDeduplicator_deduplicates_multiple_events_same_file(t *testing.T) 
 		logger := slog.Default()
 		stopCh := make(chan struct{})
 		deduplicator := NewEventDeduplicator(stopCh, handler, logger)
-		defer deduplicator.Stop()
+		defer close(stopCh)
 
 		deduplicator.QueueEvent("test.yaml")
 		deduplicator.QueueEvent("test.yaml")
@@ -58,7 +58,7 @@ func TestEventDeduplicator_processes_events_for_diferent_files(t *testing.T) {
 		logger := slog.Default()
 		stopCh := make(chan struct{})
 		deduplicator := NewEventDeduplicator(stopCh, handler, logger)
-		defer deduplicator.Stop()
+		defer close(stopCh)
 
 		deduplicator.QueueEvent("file1.yaml")
 		deduplicator.QueueEvent("file2.yaml")
@@ -90,7 +90,7 @@ func TestEventDeduplicator_resets_timer_new_event_for_same_file(t *testing.T) {
 		logger := slog.Default()
 		stopCh := make(chan struct{})
 		deduplicator := NewEventDeduplicator(stopCh, handler, logger)
-		defer deduplicator.Stop()
+		defer close(stopCh)
 
 		deduplicator.QueueEvent("test.yaml")
 		time.Sleep(100 * time.Millisecond)
@@ -116,5 +116,59 @@ func TestEventDeduplicator_resets_timer_new_event_for_same_file(t *testing.T) {
 			t.Errorf("Expected 1 processing after full debounce, got %d", result)
 		}
 
+	})
+}
+
+func TestEventDeduplicator_closes_eventCh_when_stopCh_closed(t *testing.T) {
+	synctest.Test(t, func(t *testing.T) {
+		var processedCount int
+		var mu sync.Mutex
+
+		handler := func(filename string) {
+			mu.Lock()
+			defer mu.Unlock()
+			processedCount++
+		}
+
+		logger := slog.Default()
+		stopCh := make(chan struct{})
+		deduplicator := NewEventDeduplicator(stopCh, handler, logger)
+
+		deduplicator.QueueEvent("test.yaml")
+
+		// trigger namespace controller shutdown
+		close(stopCh)
+
+		time.Sleep(50 * time.Millisecond)
+
+		select {
+		case _, ok := <-deduplicator.eventCh:
+			if ok {
+				t.Error("Expected eventCh to be closed, but it's still open")
+			}
+		case <-time.After(100 * time.Millisecond):
+			t.Error("Timeout waiting for eventCh to be closed")
+		}
+	})
+}
+
+func TestEventDeduplicator_no_double_close_panic(t *testing.T) {
+	synctest.Test(t, func(t *testing.T) {
+		handler := func(filename string) {}
+
+		logger := slog.Default()
+		stopCh := make(chan struct{})
+		deduplicator := NewEventDeduplicator(stopCh, handler, logger)
+
+		// Close stopCh (simulating NamespaceController.Stop())
+		close(stopCh)
+		time.Sleep(50 * time.Millisecond)
+
+		defer func() {
+			if r := recover(); r != nil {
+				t.Errorf("Stop() caused panic: %v", r)
+			}
+		}()
+		deduplicator.Stop()
 	})
 }

--- a/internal/nonkube/controller/event_deduplicator_test.go
+++ b/internal/nonkube/controller/event_deduplicator_test.go
@@ -141,18 +141,14 @@ func TestEventDeduplicator_closes_eventCh_when_stopCh_closed(t *testing.T) {
 
 		time.Sleep(50 * time.Millisecond)
 
-		select {
-		case _, ok := <-deduplicator.eventCh:
-			if ok {
-				t.Error("Expected eventCh to be closed, but it's still open")
-			}
-		case <-time.After(100 * time.Millisecond):
-			t.Error("Timeout waiting for eventCh to be closed")
+		_, ok := <-deduplicator.eventCh
+		if ok {
+			t.Error("Expected eventCh to be closed, but it's still open")
 		}
 	})
 }
 
-func TestEventDeduplicator_no_double_close_panic(t *testing.T) {
+func TestEventDeduplicator_no_panic_on_double_stop(t *testing.T) {
 	synctest.Test(t, func(t *testing.T) {
 		handler := func(filename string) {}
 
@@ -160,15 +156,20 @@ func TestEventDeduplicator_no_double_close_panic(t *testing.T) {
 		stopCh := make(chan struct{})
 		deduplicator := NewEventDeduplicator(stopCh, handler, logger)
 
-		// Close stopCh (simulating NamespaceController.Stop())
+		// trigger namespace controller shutdown
 		close(stopCh)
 		time.Sleep(50 * time.Millisecond)
 
 		defer func() {
 			if r := recover(); r != nil {
-				t.Errorf("Stop() caused panic: %v", r)
+				t.Errorf("Second Stop() call caused panic: %v", r)
 			}
 		}()
 		deduplicator.Stop()
+
+		_, ok := <-deduplicator.eventCh
+		if ok {
+			t.Error("Expected eventCh to be closed")
+		}
 	})
 }

--- a/internal/nonkube/controller/event_deduplicator_test.go
+++ b/internal/nonkube/controller/event_deduplicator_test.go
@@ -141,9 +141,10 @@ func TestEventDeduplicator_closes_eventCh_when_stopCh_closed(t *testing.T) {
 
 		time.Sleep(50 * time.Millisecond)
 
-		_, ok := <-deduplicator.eventCh
-		if ok {
-			t.Error("Expected eventCh to be closed, but it's still open")
+		select {
+		case <-deduplicator.done:
+		default:
+			t.Error("Expected done channel to be closed, but it's still open")
 		}
 	})
 }
@@ -167,9 +168,10 @@ func TestEventDeduplicator_no_panic_on_double_stop(t *testing.T) {
 		}()
 		deduplicator.Stop()
 
-		_, ok := <-deduplicator.eventCh
-		if ok {
-			t.Error("Expected eventCh to be closed")
+		select {
+		case <-deduplicator.done:
+		default:
+			t.Error("Expected done channel to be closed")
 		}
 	})
 }

--- a/internal/nonkube/controller/event_deduplicator_test.go
+++ b/internal/nonkube/controller/event_deduplicator_test.go
@@ -119,7 +119,7 @@ func TestEventDeduplicator_resets_timer_new_event_for_same_file(t *testing.T) {
 	})
 }
 
-func TestEventDeduplicator_closes_eventCh_when_stopCh_closed(t *testing.T) {
+func TestEventDeduplicator_closes_done_when_stopCh_closed(t *testing.T) {
 	synctest.Test(t, func(t *testing.T) {
 		var processedCount int
 		var mu sync.Mutex

